src/dal/zodiacdeep.ts

```diff
@@ -0,0 +1,126 @@
+import { database } from "../libs/database";
+import { SignData } from "../models/signData";
+import { firestore } from "firebase-admin";
+import { Signs_ZodiacDirBG } from "../models/signs";
+import { ISignRecord } from "../interfaces/signs";
+
+export async function GetArrayJSNatFromSignData(
+  sign: string
+): Promise<[{ sign: string; createdAt: string; text: string }]> {
+  try {
+    const db = database.firestore();
+    const signDataRef = db.collection("signs_ZodiacDirBG").doc(sign);
+    const doc = await signDataRef.get();
+    if (!doc.exists) {
+      console.log("No such document!");
+    } else {
+      return doc.get("data");
+    }
+  } catch (err) {
+    console.log(err);
+  }
+  return [{ sign: sign, createdAt: "", text: "" }];
+}
+
+export async function GetLastElementJSNatFromSignData(
```

Comment:
use camelCase for naming. Also rethink the name of this function. What is JSNat?